### ISSUE TITLE
Multiple output in two different loss function 

### ISSUE THREAD

**Issue by 0505li** [keras-team/keras#16322]
I used `Keras` to build a `lstm` to make prediction ，two `output layers `with it ，defined as `out1 , out2 `Now I want to combine `out1,out2` with `ytrue`, the three variables to make two `loss functions`
`My model =Model(inputs=inputs,outputs=[output1’output2])`
What I need Just like
```
define myloss(ytrue,out1,out2):
            if out1 < ytrue < out 2 :
               return k=1
            else:
               return k=0
           define loss1(ytrue,ypred):
                   return Loss1= (ypred-ytrue)+k*….
           define loss2(ytrue,ypred):
                  return Loss2= (ytrue-ypred)+k*….
          return loss1,loss2
```

**Comment by gadagashwini** [comment]
@0505li,
You can use `tf.keras.layers.Concatenate()` to concatenate output of two layers.
```
import tensorflow as tf 
import numpy as np
out1 = tf.keras.layers.Dense(8)(np.arange(10).reshape(5, 2))
out2 = tf.keras.layers.Dense(8)(np.arange(10, 20).reshape(5, 2))
concatted = tf.keras.layers.Concatenate()([x1, x2])
concatted.shape
```

`TensorShape([5, 16])`
Finally compile the model with single loss function

**Comment by 0505li** [comment]
> @0505li, You can use `tf.keras.layers.Concatenate()` to concatenate output of two layers.
> 
> ```
> import tensorflow as tf 
> import numpy as np
> out1 = tf.keras.layers.Dense(8)(np.arange(10).reshape(5, 2))
> out2 = tf.keras.layers.Dense(8)(np.arange(10, 20).reshape(5, 2))
> concatted = tf.keras.layers.Concatenate()([x1, x2])
> concatted.shape
> ```
> 
> `TensorShape([5, 16])` Finally compile the model with single loss function

Thanks for your advice.I am a begin learner for ML.In this I want to make a interval prediction to get a lower prediction and a upper prediction by using two different output layer loss functions.

**Comment by gadagashwini** [comment]
@AbhisekOmkar,

```
import tensorflow as tf 
import numpy as np
out1 = tf.keras.layers.Dense(8)(np.arange(10).reshape(5, 2))
out2 = tf.keras.layers.Dense(8)(np.arange(10, 20).reshape(5, 2))
```

While compiling your model and specifying a separate loss for each of the model's layer outputs. These losses can be strings if they are the default losses that Keras provides, or these can be callables that implement your loss function.

```
model.compile(optimizer='sgd',
              loss={
                  'dense': 'binary_crossentropy',
                  'out1': 'mse'
              })
```
```
new_model.compile(optimizer='sgd',
              loss={
                  'dense': 'binary_crossentropy',
                  'out2': 'mse'
              })
```
